src/conversion.rs

```diff
@@ -431,6 +534,13 @@ impl IntoPy<Py<PyTuple>> for () {
     }
 }
 
+impl IntoPyObject for () {
```

Comment:
Note that `IntoPy<PyObject> for ()` becomes `None`, and `IntoPy<Py<PyTuple>>` becomes `()`.

IMO this is a bit annoying, and `().into_py()` _should_ be an empty tuple 

However without specialization, we need to do some fudging in the macros to fix the `()` case as a `#[pyfunction]` return value. I'm pretty sure I know how to do this.

Reply:
To me, it would seem preferable to fix the special case of the type of the return value of functions that do not return anything instead of ingraining this into the conversion traits.

Reply:
@adamreichold I think that we're saying the same thing? At the moment the conversion traits encode this special case, however really this should just be part of the `#[pyfunction]` / `#[pymethods]` macro. (With `min_specialization` it could probably be part of the `IntoPyCallbackOutput` trait in the future, however I won't hold my breath.)

Reply:
Yes, certainly. I just wanted to echo

> IMO this is a bit annoying, and ().into_py() should be an empty tuple

which sounded like you did not fully convince yourself yet.